### ISSUE TITLE
Port FxCop rule CA1709: IdentifiersShouldBeCasedCorrectly

### ISSUE THREAD

**Issue by lgolding** [dotnet/roslyn-analyzers#445]
**Title:** Identifiers should be cased correctly

**Description:**

By convention, parameter names use camel casing and namespace, type, and member names use Pascal casing.

**Proposed analyzer:** Microsoft.ApiDesignGuidelines

**Notes:**

**Comment by mavasani** [comment]
@jinujoseph @vatsalyaagrawal - another set of customers that have requested the FxCop naming/spell check rules to be ported. I agree with @Jm-Fox here that missing porting of naming rules is currently the **biggest** migration blocker from legacy FxCop to FxCop analyzers. We need to investigate or find out which is the best spell checker library that is available for NetCore/NetStandard to unblock porting these rules.

**Comment by rosieks** [comment]
Any ETA for that rule?

**Comment by MrFoxPro** [comment]
Yeah, it is very important rule to keep up project code style. It would be nice to have support of this rule by roslyn. Could roslyn team member provide some info and maybe someone can help with PR?
